### PR TITLE
Merge dev/xml with dev/api

### PR DESCRIPTION
**Pull request by krwq**
We need to update the versions in project.jsons before final merge

### CODE REVIEW

File: src/Common/src/System/IO/StringBuilderCache.cs [hunk 1]
```diff
@@ -5,7 +5,6 @@
 
 /*============================================================
 **
-** Class:  StringBuilderCache
```
Comment (justinvp) [inline src/Common/src/System/IO/StringBuilderCache.cs]: Why?

File: src/System.Private.Xml/src/System/Xml/Base64Decoder.cs [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+            }
+            return mapBase64;
+        }
+
+        [System.Security.SecurityCritical]
+        private unsafe void Decode(char* pChars, char* pCharsEndPos,
+                             byte* pBytes, byte* pBytesEndPos,
+                             out int charsDecoded, out int bytesDecoded)
+        {
+#if DEBUG
```
Comment (justinvp) [inline src/System.Private.Xml/src/System/Xml/Base64Decoder.cs]: This ifdef isn't necessary -- `Debug.Assert` calls only remain in DEBUG builds.

File: src/System.Private.Xml/src/System/Xml/BinHexDecoder.cs [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        // Private methods
+        //
+
+        [System.Security.SecurityCritical]
+        private static unsafe void Decode(char* pChars, char* pCharsEndPos,
+                                    byte* pBytes, byte* pBytesEndPos,
+                                    ref bool hasHalfByteCached, ref byte cachedHalfByte,
+                                    out int charsDecoded, out int bytesDecoded)
+        {
+#if DEBUG
```
Comment (justinvp) [inline src/System.Private.Xml/src/System/Xml/BinHexDecoder.cs]: This ifdef isn't necessary -- `Debug.Assert` calls only remain in DEBUG builds.

File: src/Common/src/System/IO/StringBuilderCache.cs (hunk 1 above)
Comment (krwq) [inline src/Common/src/System/IO/StringBuilderCache.cs]: It must have been one of the auto-fixes we applied on the projects. Will revert.

File: src/System.Private.Xml/src/System/Xml/Base64Decoder.cs (hunk 2 above)
Comment (krwq) [inline src/System.Private.Xml/src/System/Xml/Base64Decoder.cs]: This is a port of desktop code, I'd leave those kind of improvement as a post-step

File: src/System.Private.Xml/src/System/Xml/BinHexDecoder.cs (hunk 3 above)
Comment (krwq) [inline src/System.Private.Xml/src/System/Xml/BinHexDecoder.cs]: ditto

**Comment by weshaggard** [comment]
This PR is nearly impossible to review. Can you work on squashing some of the commits into logic groups/points so we can look at them independently and try to make sense of what is happening here. 

For the 3690 files changes how many of them are additions? How many more files is XML pulling in? How many of them are tests vs product?

**Comment by krwq** [comment]
@weshaggard, We have 140 commits here including multiple both directional merges, I'm not sure how to squash this. It's possible to review it if you do folder compare: side by side corefx (easiest if you copy contracts code to one folder) and dev/xml code but it's still pretty huge and would be time consuming (we can schedule a meeting and do it in my office). IMO it only makes sense to take a look into project.jsons and .*proj files and anything not really purely code related and we can point you out to test code which is newly ported and whatever else is of interest

**Comment by krwq** [comment]
replaced by Jose's PR: https://github.com/dotnet/corefx/pull/10672

**Pull request closed without merging**